src/Core/Settings.h

```diff
@@ -339,7 +339,7 @@ struct Settings : public SettingsCollection<Settings>
     \
     M(SettingUInt64, max_memory_usage, 0, "Maximum memory usage for processing of single query. Zero means unlimited.", 0) \
     M(SettingUInt64, max_memory_usage_for_user, 0, "Maximum memory usage for processing all concurrently running queries for the user. Zero means unlimited.", 0) \
-    M(SettingUInt64, max_memory_usage_for_all_queries, 0, "Maximum memory usage for processing all concurrently running queries on the server. Zero means unlimited.", 0) \
```

Comment:
Actually `max_memory_usage_for_all_queries` can be useful to (to limit the memory usage for all queries except specific,  by overwriting `max_memory_usage_for_all_queries` to `0`), but I guess `max_memory_usage_for_user` can be used instead (but will require more attention, since there is `default` user).